src/test/kotlin/net/leanix/vsm/gitlab/broker/connector/application/ValidationServiceTest.kt

```diff
@@ -75,14 +75,34 @@ class ValidationServiceTest {
 
     @Test
     fun `it should not validate the configuration if group name is invalid`() {
+        setupService()
         every { gitlabClient.getCurrentUser() } returns getGitlabCurrentUser(true)
         every { gitlabClient.getAllGroups() } returns emptyList()
 
         assertThrows<OrgNameValidationFailed> {
-            validationService.validateConfiguration(getGitlabAssignment())
+            validationService!!.validateConfiguration(getGitlabAssignment())
         }
 
         verify(exactly = 1) { gitlabClient.getCurrentUser() }
         verify(exactly = 1) { gitlabClient.getAllGroups() }
     }
+
+    @Test
+    fun `it should not validate the user if webhook url is blank`() {
+        setupService("")
+        every { gitlabClient.getAllGroups() } returns emptyList()
+
+        assertThrows<OrgNameValidationFailed> {
+            validationService!!.validateConfiguration(getGitlabAssignment())
+        }
+
+        verify(exactly = 0) { gitlabClient.getCurrentUser() }
+        verify(exactly = 1) { gitlabClient.getAllGroups() }
+    }
+
+    private fun setupService(webhookUrl: String = "webhook-url") {
+        validationService = ValidationService(gitlabFeignClientProvider, webhookUrl)
+        validationService!!.messageSource = messageSource
+        validationService!!.loggingService = loggingService
+    }
```

Comment:
can we move this to `@BeforeEach`??

Reply:
I had moved to two nested classes, one to test with webhook url and another to test without it.